go/private/actions/compilepkg.bzl

```diff
@@ -87,6 +88,8 @@ def emit_compilepkg(
         fail("nogo must be specified if and only if out_nogo_log is specified")
     if bool(nogo) != bool(out_nogo_validation):
         fail("nogo must be specified if and only if out_nogo_validation is specified")
+    if bool(nogo) != bool(out_nogo_fix):
+        fail("nogo must be specified if and only if out_nogo_fix is specified")
```

Comment:
no, they do not need to provide a fix.
Even for the linters that have fixing rules, we do not know in advance whether it will produce fixes or not for the build target, depending on whether the target has error to fix or not. 

due to this uncertainty, conservatively, we create out_nogo_fix for each target as long as nogo is present. 

The declaration happens in archive.bzl:
    nogo = get_nogo(go)
    if nogo:
        out_facts = go.declare_file(go, name = source.library.name, ext = pre_ext + ".facts")
        out_nogo_log = go.declare_file(go, name = source.library.name, ext = pre_ext + ".nogo.log")
        out_nogo_validation = go.declare_file(go, name = source.library.name, ext = pre_ext + ".nogo")
        out_nogo_fix = go.declare_file(go, name = source.library.name, ext = pre_ext + ".nogo.fix")
    else:
        out_facts = None
        out_nogo_log = None
        out_nogo_validation = None
        out_nogo_fix = None